### PR TITLE
added support for OnError & Finally methods in FrostingTask

### PR DESCRIPTION
**Pull request by Meir017**

### CODE REVIEW

**Comment by Meir017** [comment]
@patriksvensson can you take a look at this?

File: src/Cake.Frosting.Tests/Data/Tasks/FinallyTask.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Cake.Core;
+using Cake.Core.Diagnostics;
+
+namespace Cake.Frosting.Tests.Data.Tasks
+{
+    [TaskName("Finally")]
+    public class FinallyTask : FrostingTask<ICakeContext>
```
Comment (patriksvensson) [inline src/Cake.Frosting.Tests/Data/Tasks/FinallyTask.cs]: Seal class.

File: src/Cake.Frosting.Tests/Data/Tasks/OnErrorRunCompletedTask.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Cake.Core;
+using Cake.Core.Diagnostics;
+
+namespace Cake.Frosting.Tests.Data.Tasks
+{
+    [TaskName("On-Error-Run-Completed")]
+    public class OnErrorRunCompletedTask : FrostingTask<ICakeContext>
```
Comment (patriksvensson) [inline src/Cake.Frosting.Tests/Data/Tasks/OnErrorRunCompletedTask.cs]: Seal class.

File: src/Cake.Frosting.Tests/Data/Tasks/OnErrorRunFailedTask.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Cake.Core;
+using Cake.Core.Diagnostics;
+
+namespace Cake.Frosting.Tests.Data.Tasks
+{
+    [TaskName("On-Error-Run-Failed")]
+    public class OnErrorRunFailedTask : FrostingTask<ICakeContext>
```
Comment (patriksvensson) [inline src/Cake.Frosting.Tests/Data/Tasks/OnErrorRunFailedTask.cs]: Seal class.

**Comment by Meir017** [comment]
I squashed the commits, not sure if I did it right.

**Comment by patriksvensson** [comment]
@Meir017 I can do it for you later today.

A recommendation for the next PR you do (regardless of project) is to always create a separate branch for the PR. This makes it easier to update the PR if your fork and the main fork diverge (commits added).

**Comment by patriksvensson** [comment]
@Meir017 I will need you to resubmit this PR from it's own branch since I can't rebase it without force pushing.

**Comment by patriksvensson** [comment]
@Meir017 You will need to create that branch in a clean fork (since the one you have is tainted)

**Pull request closed without merging**